pkg/utils/logger/zaplogger.go

```diff
@@ -0,0 +1,173 @@
+	lumberjack "gopkg.in/natefinch/lumberjack.v2"
+)
+
+type structuredLogger struct {
+	zapLogger *zap.SugaredLogger
+}
+
+// getZapLevel converts log level string to zapcore.Level.
+func getZapLevel(logLevel string) zapcore.Level {
+	switch lvl := strings.ToLower(os.Getenv(envLogLevel)); lvl {
+	case "debug":
+		return zapcore.DebugLevel
+	case "info":
+		return zapcore.InfoLevel
+	case "warn":
+		return zapcore.WarnLevel
+	case "error":
+		return zapcore.ErrorLevel
+	case "fatal":
+		return zapcore.FatalLevel
+	default:
+		return zapcore.DebugLevel
+	}
+}
+
+func (l *structuredLogger) Debugf(format string, args ...interface{}) {
+	l.zapLogger.Debugf(format, args...)
+}
+
+func (l *structuredLogger) Debug(format string) {
+	l.zapLogger.Desugar().Debug(format)
+}
+
+func (l *structuredLogger) Infof(format string, args ...interface{}) {
+	l.zapLogger.Infof(format, args...)
+}
+
+func (l *structuredLogger) Info(format string) {
+	l.zapLogger.Desugar().Info(format)
+}
+
+func (l *structuredLogger) Warnf(format string, args ...interface{}) {
+	l.zapLogger.Warnf(format, args...)
+}
+
+func (l *structuredLogger) Warn(format string) {
+	l.zapLogger.Desugar().Warn(format)
+}
+
+func (l *structuredLogger) Error(format string) {
+	l.zapLogger.Desugar().Error(format)
+}
+
+func (l *structuredLogger) Errorf(format string, args ...interface{}) {
+	l.zapLogger.Errorf(format, args...)
+}
+
+func (l *structuredLogger) Fatalf(format string, args ...interface{}) {
+	l.zapLogger.Fatalf(format, args...)
+}
+
+func (l *structuredLogger) Panicf(format string, args ...interface{}) {
+	l.zapLogger.Fatalf(format, args...)
+}
+
+func (l *structuredLogger) WithFields(fields Fields) Logger {
+	var f = make([]interface{}, 0)
+	for k, v := range fields {
+		f = append(f, k)
+		f = append(f, v)
+	}
+	newLogger := l.zapLogger.With(f...)
+	return &structuredLogger{newLogger}
+}
+
+func getEncoder() zapcore.Encoder {
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	return zapcore.NewJSONEncoder(encoderConfig)
+}
+
+// GetLogFileLocation returns the log file path
+func getLogFileLocation(srvName string) string {
+	switch srvName {
+	case "L-IPamD":
+		logFilePath := os.Getenv(envIPamDLogFilePath)
+		if logFilePath == "" {
+			logFilePath = defaultIPamDLogFilePath
+		}
+		return logFilePath
+	case "Plugin":
+		logFilePath := os.Getenv(envPluginFilePath)
+		if logFilePath == "" {
+			logFilePath = defaultPluginLogFilePath
+		}
+		return logFilePath
+	default:
+		return "stdout"
+	}
+}
+
+func newZapLogger(srvName string) Logger {
+	var cores []zapcore.Core
+
+	logLevel := getZapLevel(envLogLevel)
+
+	logFileLocation := getLogFileLocation(srvName)
+
+	if logFileLocation != "stdout" {
+		writer := getLogWriter(logFileLocation)
+		cores = append(cores, zapcore.NewCore(getEncoder(), writer, logLevel))
+	} else {
+		writer := zapcore.Lock(os.Stdout)
+		cores = append(cores, zapcore.NewCore(getEncoder(), writer, logLevel))
+	}
```

Comment:
I think this would be easier to read:
```go
	var writer zapcore.WriteSyncer
	if logFileLocation == "stdout" {
		writer = zapcore.Lock(os.Stdout)
	} else {
		writer = getLogWriter(logFileLocation)
	}	
	cores = append(cores, zapcore.NewCore(getEncoder(), writer, logLevel))
```